### PR TITLE
add batch property to model jinja fnction

### LINKED ISSUE #6585: [Core] New jinja `model.batch` context
### Link to the page(s) on docs.getdbt.com requiring updates

- https://docs.getdbt.com/reference/dbt-jinja-functions
- https://docs.getdbt.com/reference/dbt-jinja-functions/model

### Tell us more about this update

We've added a new `batch` object to the already existing [model](https://docs.getdbt.com/reference/dbt-jinja-functions) [dbt-jinja-functions](https://docs.getdbt.com/reference/dbt-jinja-functions/model). I'm not sure _exactly_ how this should be documented, as it a new thing on an already existing "dbt-jinja-function" and not a new "dbt-jinja-function" itself (depending on how you think about it).

## `batch` properties
The `batch` qualities available are:
- `id` (i.e. `model.batch.id`)
  - This is the "unique" id for the batch within the context of the microbatch model
- `event_time_start` (i.e. `model.batch.event_time_start`)
  - This is the start time of the batch's event time filter (inclusive)  
- `event_time_end` (i.e. `model.batch.event_time_end`)
  - This is the end time of the batch's event_time_filter (exclusive)
  
## things to call out
1. `model.batch` is only ever populated in the context of a microbatch model batch execution
   a. that is, outside of batch execution `model.batch = None` and the sub properties are unaccsible
2. Given (1a) it is recommended to guard any accessing of the underlying properties

Example of (2)
```
{% if model.batch %}
  {{ log(model.batch.id) }}
{% endif %}
```

### Reviewers/Stakeholders/SMEs

- @jtcohen6 
- @QMalcolm 
- @MichelleArk 

### Related GitHub issues

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr adds a new `batch` object to the already existing model dbt-jinja-functions.

The batch qualities available are:

id (i.e. model.batch.id)
This is the "unique" id for the batch within the context of the microbatch model
event_time_start (i.e. model.batch.event_time_start)
This is the start time of the batch's event time filter (inclusive)
event_time_end (i.e. model.batch.event_time_end)
This is the end time of the batch's event_time_filter (exclusive

Resolves #6585

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-add-batch-jinja-dbt-labs.vercel.app/reference/dbt-jinja-functions/model

### CODE REVIEW

File: website/docs/reference/dbt-jinja-functions/model.md [hunk 1]
```diff
@@ -42,6 +42,48 @@ If you're using the CLI, use [log()](/reference/dbt-jinja-functions/log) to prin
...
+### Usage notes
+
+`model.batch` is only available during the execution of a microbatch model batch. Outside of the microbatch execution, `model.batch` is `None`, and its sub-properties aren't accessible.
+
+#### Example: Safeguard access to batch properties
+
+We recommend to always check if `model.batch` is populated before accessing its properties. Use an `if` statement to ensure safe access to batch properties:
+
+```jinja
+{% if model.batch %}
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/model.md:66]: need a gut check if this snippet is correct?

File: website/docs/reference/dbt-jinja-functions/model.md [hunk 2]
```diff
@@ -42,6 +42,48 @@ If you're using the CLI, use [log()](/reference/dbt-jinja-functions/log) to prin
...
+```
+
+In this example, the `if model.batch` statement makes sure that the code only runs during a batch execution. `log()` is used to print the `batch` properties for debugging.
+
+#### Example: Log batch details
+
+This is a practical example of how you might use `model.batch` in a microbatch model to log batch details for the `batch.id`:
+
+```jinja
+{% if model.batch %}
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/model.md:80]: need a gut check if this snippet is correct?

File: website/docs/reference/dbt-jinja-functions/model.md [hunk 3]
```diff
@@ -42,6 +42,48 @@ If you're using the CLI, use [log()](/reference/dbt-jinja-functions/log) to prin
...
+## Batch properties for microbatch models
+
+From dbt Core v1.9, the model object includes a `batch` property (`model.batch`), which provides details about the current batch when executing an [incremental microbatch](/docs/build/incremental-microbatch) model. This property is only populated during the batch execution of a microbatch model.
+
+The following table describes the properties of the `batch` object. Note that dbt appends the property to the `model` and `batch` objects. 
+
+| Property | Description | Example |  
+| -------- | ----------- | ------- |
+| `id` | The unique identifier for the batch within the context of the microbatch model. | `model.batch.id` |
+| `event_time_start` | The start time of the batch's [`event_time`](/reference/resource-configs/event-time) filter (inclusive). | `model.batch.event_time_start` |
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/model.md:54]: @QMalcolm what do we mean by 'inclusive' and 'exclusive' here?

File: website/docs/reference/dbt-jinja-functions/model.md [hunk 4]
```diff
@@ -42,6 +42,48 @@ If you're using the CLI, use [log()](/reference/dbt-jinja-functions/log) to prin
...
+| `event_time_end` | The end time of the batch's `event_time` filter (exclusive). | `model.batch.event_time_end` |
+
+### Usage notes
+
+`model.batch` is only available during the execution of a microbatch model batch. Outside of the microbatch execution, `model.batch` is `None`, and its sub-properties aren't accessible.
+
+#### Example: Safeguard access to batch properties
+
+We recommend to always check if `model.batch` is populated before accessing its properties. Use an `if` statement to ensure safe access to batch properties:
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/model.md]:
```suggestion
We recommend to always check if `model.batch` is populated before accessing its properties. To do this, use an `if` statement for safe access to `batch` properties:
```

File: website/docs/reference/dbt-jinja-functions/model.md (hunk 3 above)
Comment (QMalcolm) [reply website/docs/reference/dbt-jinja-functions/model.md:54]: @mirnawong1 great question ❤️ 

### Short answer
Inclusive means the value is included in the batch, exclusive means the value is excluded from the batch

### Long answer
To think specifically about inclusivity/exclusivity means with regard to a batch, consider the batch filter statement 
```
WHERE event_time >= event_time_start AND event_time < event_time_end
```
Things to note
1. The comparison of `event_time` to `event_time_start` used `>=` with the `=` part of the operator including the `event_time_start` as part of the range. This means the usage of `event_time_start` is _inclusive_ in the range selection 
2. In the comparison of `event_time` to `event_time_end` we used `<` meaning things up to, but **not equal to**, the `event_time_end` will be selected as part of the range. This means that `event_time_end` is _exclusive_ in the range selection.

File: website/docs/reference/dbt-jinja-functions/model.md (hunk 1 above)
Comment (QMalcolm) [reply website/docs/reference/dbt-jinja-functions/model.md:66]: Looks correct to me 🙂

File: website/docs/reference/dbt-jinja-functions/model.md (hunk 2 above)
Comment (QMalcolm) [reply website/docs/reference/dbt-jinja-functions/model.md:80]: Looks correct to me 🙂

**Review by QMalcolm — APPROVED** [review]
🚀

File: website/docs/reference/dbt-jinja-functions/model.md (hunk 3 above)
Comment (mirnawong1) [reply website/docs/reference/dbt-jinja-functions/model.md:54]: oh got it, thanks so much for taking the time to clarify 🙏

File: website/docs/reference/dbt-jinja-functions/model.md [hunk 5]
```diff
@@ -20,9 +20,9 @@ To view the contents of `model` for a given model:
 
 <Tabs>
 
-<TabItem value="cli" label="CLI">
+<TabItem value="cli" label="Command line interface">
 
-If you're using the CLI, use [log()](/reference/dbt-jinja-functions/log) to print the full contents:
+If you're using the Command line interface (CLI), use [log()](/reference/dbt-jinja-functions/log) to print the full contents:
```
Comment (matthewshaver) [inline website/docs/reference/dbt-jinja-functions/model.md]:
```suggestion
If you're using the command line interface (CLI), use [log()](/reference/dbt-jinja-functions/log) to print the full contents:
```

File: website/docs/reference/dbt-jinja-functions/model.md [hunk 6]
```diff
@@ -42,6 +42,48 @@ If you're using the CLI, use [log()](/reference/dbt-jinja-functions/log) to prin
 
 </Tabs>
 
+## Batch properties for microbatch models
+
+From dbt Core v1.9, the model object includes a `batch` property (`model.batch`), which provides details about the current batch when executing an [incremental microbatch](/docs/build/incremental-microbatch) model. This property is only populated during the batch execution of a microbatch model.
```
Comment (matthewshaver) [inline website/docs/reference/dbt-jinja-functions/model.md]:
```suggestion
Starting in dbt Core v1.9, the model object includes a `batch` property (`model.batch`), which provides details about the current batch when executing an [incremental microbatch](/docs/build/incremental-microbatch) model. This property is only populated during the batch execution of a microbatch model.
```

File: website/docs/reference/dbt-jinja-functions/model.md [hunk 7]
```diff
@@ -42,6 +42,48 @@ If you're using the CLI, use [log()](/reference/dbt-jinja-functions/log) to prin
...
+
+| Property | Description | Example |  
+| -------- | ----------- | ------- |
+| `id` | The unique identifier for the batch within the context of the microbatch model. | `model.batch.id` |
+| `event_time_start` | The start time of the batch's [`event_time`](/reference/resource-configs/event-time) filter (inclusive). | `model.batch.event_time_start` |
+| `event_time_end` | The end time of the batch's `event_time` filter (exclusive). | `model.batch.event_time_end` |
+
+### Usage notes
+
+`model.batch` is only available during the execution of a microbatch model batch. Outside of the microbatch execution, `model.batch` is `None`, and its sub-properties aren't accessible.
```
Comment (matthewshaver) [inline website/docs/reference/dbt-jinja-functions/model.md:59]: We sort of state this above. Maybe we could combine the two into a callout if it's important enough

File: website/docs/reference/dbt-jinja-functions/model.md [hunk 8]
```diff
@@ -42,6 +42,48 @@ If you're using the CLI, use [log()](/reference/dbt-jinja-functions/log) to prin
...
+
+#### Example: Safeguard access to batch properties
```
Comment (matthewshaver) [inline website/docs/reference/dbt-jinja-functions/model.md]: Will the : impact the links at all? Maybe we could use the same [style we use elsewhere](https://docs.getdbt.com/docs/collaborate/explore-projects#example-of-lenses) when `Example` is in the header 
```suggestion
#### Example of safeguarding access to batch properties
```

File: website/docs/reference/dbt-jinja-functions/model.md [hunk 9]
```diff
@@ -42,6 +42,48 @@ If you're using the CLI, use [log()](/reference/dbt-jinja-functions/log) to prin
...
+{% if model.batch %}
+  {{ log(model.batch.id) }}  # Log the batch ID #
+  {{ log(model.batch.event_time_start) }}  # Log the start time of the batch #
+  {{ log(model.batch.event_time_end) }}  # Log the end time of the batch #
+{% endif %}
+```
+
+In this example, the `if model.batch` statement makes sure that the code only runs during a batch execution. `log()` is used to print the `batch` properties for debugging.
+
+#### Example: Log batch details
```
Comment (matthewshaver) [inline website/docs/reference/dbt-jinja-functions/model.md]:
```suggestion
#### Example of log batch details
```

**Pull request merged**